app/src/main/java/com/deitel/tipcalculator/MainActivity.java

```diff
@@ -2,116 +2,177 @@
+        peopleEditText.addTextChangedListener(peopleEditTextWatcher);
+
+
+
+    }
+
+    // calculate and display tip and total amounts
+    private void calculate() {
+        // format percent and display in percentTextView
+        percentTextView.setText(percentFormat.format(percent));
+
+        // calculate the tip and total
+        double tip = billAmount * percent;
+        double total = billAmount + tip;
+        double total_ppl = total / peopleAmount; //zin
+
+
+        // display tip and total formatted as currency
+        tipTextView.setText(currencyFormat.format(tip));
+        totalTextView.setText(currencyFormat.format(total));
+        eachTextView.setText(currencyFormat.format(total_ppl)); // zin
+
+    }
+
+    // listener object for the SeekBar's progress changed events
+    private final OnSeekBarChangeListener seekBarListener =
+            new OnSeekBarChangeListener() {
+                // update percent, then call calculate
+                @Override
+                public void onProgressChanged(SeekBar seekBar, int progress,
+                                              boolean fromUser) {
+                    percent = progress / 100.0; // set percent based on progress
+                    calculate(); // calculate and display tip and total
+                }
+
+                @Override
+                public void onStartTrackingTouch(SeekBar seekBar) {
+                }
+
+                @Override
+                public void onStopTrackingTouch(SeekBar seekBar) {
+                }
+            };
+
+    // listener object for the EditText's text-changed events
+    private final TextWatcher amountEditTextWatcher = new TextWatcher() {
+        // called when the user modifies the bill amount
+        @Override
+        public void onTextChanged(CharSequence s, int start,
+                                  int before, int count) {
+
+            try { // get bill amount and display currency formatted value
+                billAmount = Double.parseDouble(s.toString()) / 100.0;
+                amountTextView.setText(currencyFormat.format(billAmount));
+            } catch (NumberFormatException e) { // if s is empty or non-numeric
+                amountTextView.setText("");
+                billAmount = 0.0;
+            }
+
+            calculate(); // update the tip and total TextViews
+        }
+
+
+        @Override
+        public void afterTextChanged(Editable s) {
+        }
+
+        @Override
+        public void beforeTextChanged(
+                CharSequence s, int start, int count, int after) {
+        }
+    };
+
+    // listener object for the EditText's text-changed events // zin
+
+    private final TextWatcher peopleEditTextWatcher = new TextWatcher() {
+
+        //called when the user change the amount of party
+        @Override
+        public void beforeTextChanged(CharSequence s, int start, int count, int after) {
+
+            try {
```

Comment:
Review your logic in this part. You need to get the value entered in _peopleEditText_ (as a string) and update the _peopleAmount_  variable to reflect the new value from the EditText box. Pay attention to converting from a String to an Integer (use **Int.parseString**) and to making sure the default value for party size is at least 1.